src/store/songs/actions.js

```diff
@@ -0,0 +1,16 @@
+import { CHANGE_VOTE_FAIL } from './constants'
+
+export const changeVote = song => (dispatch, getState, getFirebase) =>
+  async value => {
+    const firebase = getFirebase()
+    const votes = song.votes || 0
+    const newVotes = votes + value
+    try {
+      firebase.update(`/songs/${song.id}`, { votes: newVotes })
```

Comment:
this `try` block may be a reasonable place to put `localStorage` vote tracking logic